src/config/file.rs

```diff
@@ -100,15 +100,21 @@ impl ConfigFile {
         }
     }
 
-    pub(crate) fn get_profile_names(config: &str) -> ConfigFileResult<Vec<String>> {
-        let mut profiles: Vec<String> = Vec::new();
+    pub(crate) fn get_profile_details(config: &str) -> ConfigFileResult<Vec<ProfileDetails>> {
+        let mut profiles: Vec<ProfileDetails> = Vec::new();
         if !config.is_empty() {
             let config_file: ConfigFile = serde_yaml::from_str(&config)?;
             profiles = config_file
                 .profiles
-                .keys()
-                .cloned()
-                .collect::<Vec<String>>();
+                .iter()
+                .map(|(k, v)| ProfileDetails {
+                    api_key: v.api_key.clone(),
+                    description: v.description.clone(),
+                    environment: v.environment.clone(),
```

Comment:
You probably want to add environment and project entries in the config file template, even if they're commented out to start. That template is used when editing the config for the first time and serves as a good place to document configuration items.